src/docs/widgets.md

```diff
@@ -0,0 +1,156 @@
+---
+title: Widgets
+---
+
+# Widgets
+
+A widget is a part that will be displayed within the Theia workbench, e.g. a view or an editor. Examples for existing widgets in Theia are the file explorer, the code editor or the problems view. By contributing custom widgets, you can place your own custom UI in a Theia-based application. The Theia workbench provides a frame so that you can embed custom UI into a so-called widget. Your custom UI will behave the same as other widgets in terms of the window layouting including the title tab, resizing, dragging and opening/closing (see screenshot below). 
+
+<img src="/widget-example.gif" alt="Widget Example" style="max-width: 525px">
+
+Furthermore, a widget will receive events from the surrounding workbench, e.g. on application start, on resize or on detach. The implementation of the actual content of a widget, which is rendered in the provided frame is completely up to you, though. As an example, you can implement some custom UI using React within a widget.
+
+In a nutshell, a widget is a frame to embed some custom (HTML-based) UI into the Theia workbench (see diagram below)
+
+<img src="/widget-architecture.png" alt="Widget Achitecture" style="max-width: 525px">
+
+In this article we will describe how to contribute a custom widget to the Theia workbench. We will focus on a simple view (in contrast to an editor) and use React to implement the UI. 
+
+If you are not yet familiar with contribution points in Theia or the use of dependency injection, please consider this guide on [Services and Contributions](https://theia-ide.org/docs/services_and_contributions/).
+
+If you would like to have a look at the example code, please use the [Theia extension generator](https://github.com/eclipse-theia/generator-theia-extension). Install the generator, select the “Widget” example and enter “MyWidget” as a name for the extension.
+
+## Contributing a widget (a view)
+
+Contributing a widget, in our example a view consists of three components:
+
+<ul>
+<li>The actual <b>widget</b>, responsible for:
+    <ul>
+    <li>Setting base parameters, such as an ID, a label and an icon</li>
+    <li>Creating the actual UI and its behavior</li>
+    <li>Reacting to lifecycle events such as `onUpdateRequest` or `onResize`</li>
+    </ul>
+</li>
+<li>A <b>widget factory</b>, responsible for instantiating the widget</li>
+<li>A <b>widget contribution</b>, responsible for wiring the view with the Theia workbench so that the widget can be opened from within the Theia workbench, e.g. via the views menus</li>
+</ul>
```

Comment:
minor: I believe a standard list (non-html) would work here as well and is simpler.

Reply:
True! However, the rendering of nested lists is apparently broken. I have an item on my TODO list to report this issue, need to create a reproducer. For this reason, I switched to HTML, cause it works.

Reply:
https://github.com/eclipse-theia/theia-website/issues/221